packages/ra-core/src/controller/useListController.ts

```diff
@@ -177,7 +177,7 @@ const useListController = <RecordType extends Record = Record>(
         get(state.admin.resources, [resource, 'list', 'ids'], [])
     );
     const defaultTotal = useSelector((state: ReduxState): number =>
-        get(state.admin.resources, [resource, 'list', 'total'], 0)
+        get(state.admin.resources, [resource, 'list', 'total'])
```

Comment:
If you don't pass 0, the result could be "undefined". It could lead to errors if the sub-components don't manage this use case.

Reply:
Normally they all do. That's why I fixed the Pagination component. 

As for user land components, yes this is a BC break, albeit minor in my opinion. And it's necessary to fix #4601